bundles/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/internal/PentairParser.java

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+*/
+package org.openhab.binding.pentair.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implements the thread to read and parse the input stream. Once a packet can be identified, it locates the
+ * representative sending Thing and dispositions the packet so it can be further processed.
+ *
+ * @author Jeff James - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class PentairParser implements Runnable {
+    private final Logger logger = LoggerFactory.getLogger(PentairParser.class);
+
+    public static final int MAXPENTAIRPACKET = 50;
+
+    private enum ParserState {
+        WAIT_STARTOFPACKET,
+        CMD_PENTAIR,
+        CMD_INTELLICHLOR
+    };
+
+    @Nullable
+    InputStream reader;
+
+    public void setInputStream(InputStream reader) {
+        // this.reader = Optional.of(reader);
+        this.reader = reader;
+    }
+
+    /**
+     * Callback interface when a packet is received
+     */
+    public interface CallbackPentairParser {
+        public void onPentairPacket(PentairPacket p);
+
+        public void onIntelliChlorPacket(PentairIntelliChlorPacket p);
+    };
+
+    private Optional<CallbackPentairParser> callback = Optional.empty();
+
+    public void setCallback(CallbackPentairParser cb) {
+        callback = Optional.of(cb);
+    }
+
+    private int getByte() throws IOException, InterruptedException {
+        int c = 0;
+
+        InputStream reader = Objects.requireNonNull(this.reader, "Reader has not been initialized.");
+
+        do {
+            c = reader.read();
+            if (c != -1) {
+                break;
+            }
+            Thread.sleep(600);
```

Comment:
`read()` is supposed to block until there is input or return -1 if it is the end of the stream. Why do you keep trying to read after it returns -1?

Reply:
I must have read the documentation incorrectly when this was implemented. I removed the loop.